shark_turbine/kernel/ops/wave_ops.py

```diff
@@ -374,6 +378,20 @@ def index(self, value: Any):
         else:
             raise ValueError("Index must be a dict")
 
+    def pre_expansion(self, constraints: list["Constraint"]) -> None:
+        """
+        Hook for pre-expansion operations. This is called before the arguments
+        of the node are expanded.
+        """
+        pass
+
+    def post_expansion(self, constraints: list["Constraint"]) -> None:
```

Comment:
This is kind of nice, I can it see it being very useful in futureto propagate types and shape information for post and pre-expansion :)